pkg/kyverno/common/fetch.go

```diff
@@ -272,6 +272,33 @@ func convertResourceToUnstructured(resourceYaml []byte) (*unstructured.Unstructu
 	return resource, nil
 }
 
+// GetPatchedResource converts raw bytes to unstructured object
+func GetPatchedResource(patchResourceBytes []byte) (patchedResource unstructured.Unstructured, err error) {
```

Comment:
Is this function required ?
This function is very similar to `GetResource`. Whats the difference ? Can we reuse the same method with minor changes if needed ?

Reply:
> 
> 
> Is this function required ?
> This function is very similar to `GetResource`. Whats the difference ? Can we reuse the same method with minor changes if needed ?

Resource can have more than one path, therefore GetResource return one or more than one resource. But, patchedResource path can be only one at a time.  So, it return one one patched resource. 
I tried it. I did small changes and it seems to work,

```
func GetPatchedResource(patchResourceBytes []byte) (patchedResource unstructured.Unstructured, err error) {
	x, err := GetResource(patchResourceBytes)
	y := x[0]
	patchedResource = *y
	return patchedResource, nil
}
```